### ISSUE TITLE
weighted loss layer for classification.

### ISSUE THREAD

**Issue by daisyron** [davisking/dlib#1036]
Hi,

I looked in the code for a weighted loss layer for classification - expected to see loss_multiclass_log_weighted.

I have an un-balanced data set for training (much more examples of one class compared to the other), and I thought to weight the rare class examples with higher loss....

Is there a way to extend the loss_multiclass_log layer to support such a need?

Is there another loss layer that can be used for such a purpose?

Thanks in advance,

Ron.

**Comment by davisking** [comment]
There isn't anything in dlib for this.  However, I would just create balanced mini-batches by oversampling as appropriate from any classes that are under represented in your training data.  Have you tried that and found that it isn't as good as a direct weighting in the loss?  I would expect both methods to be the same.

**Comment by daisyron** [comment]
Thanks for your answer. I'll try your suggestion.
Regards,
Ron.
